packages/core/__tests__/interpreter/takeSync.js

```diff
@@ -643,3 +654,29 @@ test('put causing sync dispatch response in store subscriber', () => {
     expect(actual).toEqual(['a', 'b'])
   })
 })
+
+test('action dispatched in root saga should get scheduled and taken by a "sibling" take', () => {
+  const reducer = (state, action) => {
+    if (!state) return []
+
+    return state.concat(action.type)
+  }
+
+  const middleware = sagaMiddleware()
+  const store = createStore(reducer, applyMiddleware(middleware))
+
+  function* root() {
+    yield all([
+      put({ type: 'FIRST' }),
+      takeEvery('FIRST', function*() {
```

Comment:
what about such case, where we use use `proc` internally for iterable
```js
function* root() {
    yield all([
      put({ type: 'FIRST' }),
      function*() {
        yield take('FIRST')
        yield put({ type: 'SECOND' })
      },
    ])
  }

```

Reply:
aren't they the same? `takeEvery` is also using proc internally, as it's yielding `fork` which is wrapped with proc.

Reply:
there is no `takeEvery`, if you add this test you will see that it fails

Reply:
good catch! should be handled as well

Reply:
actually ur given snippet doesnt work, as it yields a generator and not the iterator, even if we correct it to:
```js
function* root() {
    yield all([
      put({ type: 'FIRST' }),
      (function*() {
        yield take('FIRST')
        yield put({ type: 'SECOND' })
      })(),
    ])
  }
```

the second yielded item in all is equivalent to `call` effect (kinda) and thus scheduler doesn't apply to it

or did I miss something?